### DISCUSSION TITLE
Incompatibility of local domain creation and ssl with the new Windows update

### DISCUSSION THREAD

**Discussion by asadister** [leokhoa/laragon#601] [General]
In the new Windows update, Local SSL, it cannot change the host file content and it also shows the certificate as invalid.

![ssl](https://github.com/leokhoa/laragon/assets/77143123/c0eea519-639f-4aaa-87a1-bba211db7a39)
<img href="https://github.com/leokhoa/laragon/assets/77143123/c0eea519-639f-4aaa-87a1-bba211db7a39">

**Comment by Raimun2000** [comment]
Did you try removing the exception?

**Comment by Pen-y-Fan** [comment]
I've had the same, don't know why. My fix was to re-add SLL to the trusted store: Apache > SSL > Add laragon.crt to Trusted Store. 

See my blog [How to set up Laragon on a new Windows computer (part 4) - Enable security](https://pen-y-fan.github.io/2022/11/10/how-to-set-up-laragon-on-a-new-windows-computer-part-4/) for more details.

**Comment by jeffegiovani** [comment]
For me, only this solved -> https://www.youtube.com/watch?v=QnKD-p0tTpw

**Reply by RogerioAmaral** [reply]
Yes!! For me, only this solved:  https://www.youtube.com/watch?v=QnKD-p0tTpw
Tks @jeffegiovani

**Reply by DamChtlv** [reply]
@jeffegiovani Thanks a lot! Had the issue too just recently and it works :)

**Reply by Tiebob** [reply]
Thanks! It works

**Comment by KingWebsites** [comment]
None of these fixes are working for me unfortuately, in both Firefox and Chrome.

Chrome: NET::ERR_CERT_INVALID
Firefox:  SEC_ERROR_INADEQUATE_KEY_USAGE

I've changed the keyUsage setting in the 3 openssl.conf files to:
keyUsage = nonRepudiation, digitalSignature, keyEncipherment

and I've re-added laragon.crt to Trusted Store.

Are there any other options available? (Win10)

**Reply by KingWebsites** [reply]
Update. I seem to have fixed this for now.

I closed both browsers.
Laragon / Menu / Apache / SSL / Enabled (untick).
Laragon / Menu / Apache / Stop Apache.
Laragon / Menu / Apache / SSL / Enabled (tick).
Laragon / Menu / Apache / Start Apache.
Laragon / Menu / Apache / SSL / Add laragon.crt to Trust Store.
Open browsers.

**Comment by bcsirt** [comment]
Use .localhost

*.localhost is reserved for local development (RFC 6761)

Natively recognized by browsers

No conflicts with real domains

Recommended Laragon Configuration

Modify domains in Laragon:

Menu → Preferences → General

Change {name}.test to {name}.localhost

Generate wildcard certificate:

# Install mkcert ( on windows )
choco install mkcert

# Install mkcert if not already done

mkcert -install

# Create wildcard certificate for localhost

mkcert "*.localhost" localhost 127.0.0.1 ::1

Regards

[BCSIRT](https://bcsirt.fr)
